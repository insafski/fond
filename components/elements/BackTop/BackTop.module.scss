.backtop {
	width: 3.5rem;
	height: 3.5rem;
	background-color: rgba(137, 43, 226, 0.199);
	border-radius: 50%;
	position: fixed;
	right: 3rem;
	bottom: 3rem;
	outline: none;
	opacity: 1;
	transition: opacity 0.3s ease-in 0s;

	&:focus {
		outline: none;
	}
	&.hide {
		opacity: 0;
	}
}
