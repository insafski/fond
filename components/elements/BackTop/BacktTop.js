import { animateScroll as scroll } from "react-scroll";
import { useEffect } from "react";
import { useToggle } from "ahooks";

import styles from  "./BackTop.module.scss";

export default function BackTop() {

	const [state, { toggle }] = useToggle(false);

	useEffect(() => {
		window.addEventListener("scroll", handleScroll);
		return () => window.removeEventListener("scroll", handleScroll);
	});

	const handleScroll = () => {
		window.scrollY > 1000 ? toggle(true) : toggle(false);
	};

	return (
		<button className={styles.backTop} onClick={() => scroll.scrollToTop()} style={!state ? {display: "none"}: {}}>
			<i className={"hm-forward"}></i>
		</button>
	);
}
