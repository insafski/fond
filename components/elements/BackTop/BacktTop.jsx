import React, { useEffect } from "react";
import { animateScroll as scroll } from "react-scroll";
import { useToggle } from "ahooks";
import cx from "classnames";

import styles from "./BackTop.module.scss";

export default function BackTop() {
	const [state, { toggle }] = useToggle(false);

	function handleScroll() {
		toggle(window.scrollY > 1000);
	}

	useEffect(() => {
		window.addEventListener("scroll", handleScroll);
		// TODO: Need to use useScroll

		return () => window.removeEventListener("scroll", handleScroll);
	});

	return (
		<button className={cx(styles.backtop, { [styles.hide]: !state })} onClick={() => scroll.scrollToTop()}>
			<i className={"fscch-up-open"}></i>
		</button>
	);
}
