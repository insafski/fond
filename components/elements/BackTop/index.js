import BacktTop from "./BacktTop";

export default BacktTop;
