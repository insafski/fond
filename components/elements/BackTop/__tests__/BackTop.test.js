import React from "react";
import renderer from "react-test-renderer";

import BackTop from "..";

describe("BackTop", () => {
	it("renders correctly", () => {
		const tree = renderer
			.create(<BackTop />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
