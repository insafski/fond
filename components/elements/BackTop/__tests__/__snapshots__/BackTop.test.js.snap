// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackTop renders correctly 1`] = `
<button
  className="backtop hide"
  onClick={[Function]}
>
  <i
    className="hm-forward"
  />
</button>
`;
